batch_processing.sh

```diff
@@ -93,6 +93,8 @@ sct_process_segmentation -i t2_seg.nii.gz -vert 2:3 -o csa_c2c3.csv
 sct_detect_pmj -i t2.nii.gz -c t2 -qc "$SCT_BP_QC_FOLDER" 
 # Compute cross-section area at 60 mm from PMJ averaged on a 30 mm extent
 sct_process_segmentation -i t2_seg.nii.gz -pmj t2_pmj.nii.gz -pmj-distance 60 -pmj-extent 30 -qc "$SCT_BP_QC_FOLDER" -qc-image t2.nii.gz -o csa_pmj.csv
+# Compute morphometrics in PAM50 anatomical dimensions
+sct_process_segmentation -i t2_seg.nii.gz -vertfile t2_seg_labeled.nii.gz -perslice 1 -normalize PAM50 -o csa_pam50.csv
```

Comment:
It looks like the `pytest` tests now pass. However, the `batch_processing.sh` test fails, due to this mentioning `-normalize PAM50` instead of `-normalize-PAM50`.

```suggestion
# Compute morphometrics in PAM50 anatomical dimensions
sct_process_segmentation -i t2_seg.nii.gz -vertfile t2_seg_labeled.nii.gz -perslice 1 -normalize PAM50 -o csa_pam50.csv
```
```suggestion
# Compute morphometrics in PAM50 anatomical dimensions
sct_process_segmentation -i t2_seg.nii.gz -vertfile t2_seg_labeled.nii.gz -perslice 1 -normalize-PAM50 -o csa_pam50.csv
```

Reply:
Fixed in [99caabe](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3977/commits/99caabe4ad9bd5e145cc10ca5d148d70d6b0c6c8).

Reply:
It looks like this is throwing a new error now:

```
# Display results (to easily compare integrity across SCT versions)
# ===========================================================================================
set +v
~~~
Version:         git-HEAD-fd16b0e4c927eeabb282304998e38ddb797b8206
Ran on:          Linux fv-az133-476 5.15.0-1031-azure
Duration:        0hrs 18min 45sec
---
t2/csa_c2c3.csv:         73.87711295363036  [Row 1, MEAN(area)]
t2/csa_pmj.csv:          73.89298021190447  [Row 1, MEAN(area)]
Traceback (most recent call last):
  File "testing/batch_processing/test_batch_processing.py", line 68, in <module>
    display_batch_processing_results()
  File "testing/batch_processing/test_batch_processing.py", line 63, in display_batch_processing_results
    value = get_csv_float_value(OUTPUT_DIR/csv_filepath, row, column)
  File "testing/batch_processing/test_batch_processing.py", line 28, in get_csv_float_value
    with open(csv_filepath, newline='') as csvfile:
FileNotFoundError: [Errno 2] No such file or directory: '/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/data/sct_example_data/t2/csa_pam50_PAM50.csv'
```

I guess because we got rid of the extra `_PAM50.csv` file, we need to just use the normal output file (`csa_pam50.csv`).

Reply:
Done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3977/commits/1a44606f4be4a7ff34def3b23621f49d35c1d151.

Reply:
Ah shoot, we have yet another error:

```
Traceback (most recent call last):
  File "testing/batch_processing/test_batch_processing.py", line 68, in <module>
    display_batch_processing_results()
  File "testing/batch_processing/test_batch_processing.py", line 63, in display_batch_processing_results
    value = get_csv_float_value(OUTPUT_DIR/csv_filepath, row, column)
  File "testing/batch_processing/test_batch_processing.py", line 31, in get_csv_float_value
    return float(value)
ValueError: could not convert string to float: ''
batch_processing.sh: line 270: ./venv_sct/Scripts/python.exe: No such file or directory
```

The problem is because the cached .csv file starts at slice 80:

![image](https://user-images.githubusercontent.com/16181459/215612842-d476d9da-1d7d-4896-986c-b21f02867e27.png)

While the generated .csv file starts at slice 0:

![image](https://user-images.githubusercontent.com/16181459/215612918-118c30ea-1962-4230-a447-14c2f21a4244.png)

Meaning that we index different rows for both the cached and generated csv files.

Reply:
We can fix this by either adding rows to the cached file, or changing the code to exclude PAM50 slices without vertebral levels...

Of course, all of this is kind of moot if we end up implementing https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3977#discussion_r1089294999.

Reply:
if we change the test for batch processing, on which row to check, would it work?

Reply:
Any row with metrics (so, vert levels 2,3,4,5) should work. We just need to make sure that the cached file and the generated file have matching rows.

Reply:
hmm, and how come the cached file is different? How is it generated?

Reply:
I'm not sure, since I think it was uploaded by you in https://github.com/spinalcordtoolbox/spinalcordtoolbox/commit/fff1bdf9b3d2996876a7e5fcb496b0d1de68691a. :sweat_smile: 

I think the best plan is to figure out what we want the default `csa_pam50.csv` to look like:

1. Only slices present in the vertebral segmentation for `im`
2. Only slices present in the vertebral segmentation for `PAM50_levels.nii.gz`
3. _All_ slices present in `PAM50_levels.nii.gz`

Then, we update the behavior in the code, and regenerate the csv file from scratch by running the relevant section in `batch_processing.sh`.

Reply:
Personally I think I'm still in favor of having just the `im` slices by default, then using the PAM50 slices for `-v 2`. 

If we make that change, we'd have to regenerate the cached file anyway?

Reply:
> Personally I think I'm still in favor of having just the `im` slices by default, then using the PAM50 slices for `-v 2`.
> 
> If we make that change, we'd have to regenerate the cached file anyway?

Let's go with that! In that case, it gives us flexibility for further features using that

Reply:
done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3977/commits/c7459d7d94f6b20f0709d662152c7eaac631d7c5

Now we can adjust the cached file!

Reply:
just to be sure, how can I create the cached file? I can run batch_processing and put that file as a chached file?

Reply:
Yes exactly!

(Normally what I do to save a bit of time is only run the commands that are necessary, and skip the T1/dMRI/fMRI processing.